node-api/backend/genesis.go

```diff
@@ -26,12 +26,67 @@ import (
 	"github.com/berachain/beacon-kit/primitives/math"
 )
 
-// GetGenesis returns the genesis state of the beacon chain.
-func (b Backend) GenesisValidatorsRoot(slot math.Slot) (common.Root, error) {
-	// needs genesis_time and genesis_fork_version
-	st, _, err := b.stateFromSlot(slot)
+// GetGenesisValidatorsRoot returns the genesis validators root of the beacon chain.
+func (b *Backend) GenesisValidatorsRoot() (common.Root, error) {
+	// Fast path: read lock for checking cached value
+	b.genesisValidatorsRootMu.RLock()
+	if b.genesisValidatorsRoot != (common.Root{}) {
+		root := b.genesisValidatorsRoot
+		b.genesisValidatorsRootMu.RUnlock()
+		return root, nil
+	}
+	b.genesisValidatorsRootMu.RUnlock()
+
+	// Slow path: write lock for initialization
+	b.genesisValidatorsRootMu.Lock()
+	defer b.genesisValidatorsRootMu.Unlock()
+
+	// Double check after acquiring write lock
+	if b.genesisValidatorsRoot != (common.Root{}) {
+		return b.genesisValidatorsRoot, nil
+	}
+
+	// If not cached, read state from the genesis slot
```

Comment:
reading from genesis slot is not possible

Reply:
0 -> implies "head" in comet

Reply:
As discussed on the call, keeping it as it is - fetching the state from slot 0. As we don't have validatorsRoot value in the genesis file.